### PR TITLE
feat: create kb article for github raw proxies in idp

### PR DESCRIPTION
**Pull request by rssnyder**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

- Creating a new folder for sharing proxy configurations
- Added an article to guide users in creating a backend proxy for raw github files

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Review by OrkoHunter — COMMENTED** [review]
@deba-harness please review and merge. I do think this deserves to be mentioned from the Dynamic Workflow Picker page in our docs. Just putting in KB section is hiding it a bit too much.

**Comment by OrkoHunter** [comment]
Thank you Riley! First non-IDP team contributor to our docs :')

File: kb/internal-developer-portal/articles/backend-proxies/github-raw.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: GitHub Raw Files
+description: Retrieve raw JSON files from github repositories
```
Comment (brandon-harness) [inline kb/internal-developer-portal/articles/backend-proxies/github-raw.md:3]: GitHub

File: kb/internal-developer-portal/articles/backend-proxies/github-raw.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+---
+
+# Overview
+
+Instead of using an API to retrieve JSON data with a backend proxy, we can store JSON files in github and use an HTTP call to get the raw file back as pure JSON.
```
Comment (brandon-harness) [inline kb/internal-developer-portal/articles/backend-proxies/github-raw.md:8]: GitHub

File: kb/internal-developer-portal/articles/backend-proxies/github-raw.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        /api/proxy/github-raw/?: /
+      headers:
+        Authorization: Bearer ${PROXY_GITHUB_TOKEN}
+```
+
+What we are doing here is:
+
+1. Declaring a proxy called `github-raw`
+2. Targeting a base URL of `https://raw.githubusercontent.com`
+3. Rewriting any request to `/api/proxy/github-raw/+` to `https://raw.githubusercontent.com/+`
```
Comment (brandon-harness) [inline kb/internal-developer-portal/articles/backend-proxies/github-raw.md:33]: Should this sentence be 'Rewriting any request FROM ..... to ......

should this '/api/proxy/github-raw/+' be '/api/proxy/github-raw/?'

should this 'https://raw.githubusercontent.com/+' be 'https://raw.githubusercontent.com'

File: kb/internal-developer-portal/articles/backend-proxies/github-raw.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+4. Passing an authentication header using the variable `PROXY_GITHUB_TOKEN`
+
+We will need to create a variable below called `PROXY_GITHUB_TOKEN` and select the Harness account secret that holds the value. This need to be a GitHub PAT that can resolve files from your target repositories.
```
Comment (brandon-harness) [inline kb/internal-developer-portal/articles/backend-proxies/github-raw.md:36]: needs

**Review by brandon-harness — APPROVED** [review]

**Pull request merged**